src/daft-scan/src/glob.rs

```diff
@@ -233,12 +240,12 @@ impl GlobScanOperator {
             }
             false => schema.expect("Schema must be provided if infer_schema is false"),
         };
-
         Ok(Self {
             glob_paths: glob_paths.iter().map(|s| s.to_string()).collect(),
             file_format_config,
             schema,
             storage_config,
+            parquet_metadata,
```

Comment:
I believe that this would pass the same `parquet_metadata` to different files which would be incorrect! When we glob, we typically only read 1 parquet metadata to infer the schema. But the rest of the globs and parquet metadata fetches actually occur [here](https://github.com/Eventual-Inc/Daft/blob/9585e71d6664ce0b8c8eafa82bea046ba5199ed3/src/daft-scan/src/scan_task_iters.rs#L175 ) where we split row groups. Note: if we have more than a certain number of files, we skip row group splitting!

Reply:
this should be fixed now to properly get the metadata per file.

I'll have to take a closer look at the split rowgroup function to see when that's being invoked.